src/PrestaShopBundle/EventListener/Context/Admin/ShopContextListener.php

```diff
@@ -116,4 +127,13 @@ private function getShopConstraintFromCookie(): ?ShopConstraint
 
         return null;
     }
+
+    private function changeShopContext(string $shopContextUrlParameter)
```

Comment:
I'm not sure this is the proper format for shop context, it's supposed to be something like `s-1` for single shop, for shop group I'm not sure but it needs to be investigated, maybe by checking the recent Multistore header in its links And for all shops I know even less the expected format

But what's for sure is that it's not just storing the ID itself, and also I believe we need to perform a redirection after setting this value, so this method could return a redirect response when relevant

Or at least it was needed for the legacy switch 🤔 but maybe since we handle this switch early enough the redirection is not useful anymore, that I'm not really sure of Maybe it's safer to do the same for now until we refacto this properly